core/src/main/java/hudson/model/Slave.java

```diff
@@ -373,7 +375,7 @@ private URLConnection connect() throws IOException {
             return res.openConnection();
         }
 
-        public URL getURL() throws MalformedURLException {
+        public URL getURL() throws IOException {
```

Comment:
>  I cannot imagine many reasons for plugin code to be calling it

I'm always happy to Google for you

* https://github.com/TheIndifferent/jenkins-cloud-vbox-host-plugin/blob/333ef2137ed2edea58af9cf6a78d60943fd2f108/src/main/java/hudson/plugins/cloud/vbox/machines/VMSlaveLauncher.java#L168
* https://github.com/jenkinsci/foreman-node-sharing-plugin/blob/cbf2ac5321e66bbf862d50b0b600ab9db52bc553/src/test/java/com/redhat/foreman/launcher/ForemanDummyComputerLauncherFactory.java#L25